### PR TITLE
Add binary search to L0_quick_search

### PR DESCRIPTION
**Pull request by nv-braf**
Adding  `--latency-budget 10` to **L0_quick_search** will enable a binary search.

This exposed some bugs in the code around cases where PA doesn't return a result and we add `None` to the ConcurrencySearch's RCM list.

I've updated the code to check and deal with None.

### CODE REVIEW

File: model_analyzer/result/concurrency_search.py [hunk 1]
```diff
@@ -111,31 +113,56 @@ def _has_objective_gain_saturated(self) -> bool:
...
         else:
             return False
 
     def _at_constraint_failure_boundary(self, index: int) -> bool:
-        at_failure_boundary = not self._run_config_measurements[
+        if not self._run_config_measurements[
+                index] or not self._run_config_measurements[index - 1]:
+            return False
+
+        at_failure_boundary = not self._run_config_measurements[  # type: ignore
```
Comment (tgerdesnv) [inline model_analyzer/result/concurrency_search.py:162]: are 2 type ignores needed? Makes this hard to read

**Review by tgerdesnv — COMMENTED** [review]
I'm glad to hear testing caught issues!

File: model_analyzer/result/concurrency_search.py (hunk 1 above)
Comment (nv-braf) [reply model_analyzer/result/concurrency_search.py:162]: Grrr...yes, it's really annoying, but I can't figure out any other way to do this.

Comment (tgerdesnv) [reply model_analyzer/result/concurrency_search.py:162]: optional measurements have always caused us problems. These type hints point to part of our design that we should try to clean up

Comment (nv-braf) [reply model_analyzer/result/concurrency_search.py:162]: Well, tell PA to stop returning no measurement and this wouldn't need to be optional!

**Pull request merged**